_pytest/main.py

```diff
@@ -388,7 +390,11 @@ def _perform_collect(self, args, genitems):
         else:
             if rep.passed:
                 for node in rep.result:
-                    self.items.extend(self.genitems(node))
+                    items_list = []
```

Comment:
```
items = self.genitems(node)
```
returns an error that you can't assign generator object

Reply:
Oh right, but because my code sample shadows the other `items` in the same scope:

```python
            self.items = items = []  # note "items" here
            ...
            if rep.passed:
                for node in rep.result:
                    items = self.genitems(node)  # "items" shadows the previous one
                    self.items.extend(items)
            return items  # oops!
```

Checking the code now, it doesn't seem to make sense to create that alias to `self.items` anymore (probably when it was first written it did, but overtime it lost its usefulness). I suggest to remove the alias altogether:

```python
            self.items = []  
            ...
            if rep.passed:
                for node in rep.result:
                    items = self.genitems(node)
                    if node.fspath in self._input_lines:
                        ...    
                    self.items.extend(items)
            return self.items
```